### PR TITLE
Complete Add/Set Command, JUnit Tests

### LINKED ISSUE #4: Add Components in Series/Parallel
Ability to add components to a template configuration (#3) in series/parallel. Requires #5.

Based on the [User Guide section](https://github.com/AY2021S1-CS2113T-W13-3/tp/blob/master/docs/UserGuide.md#add-component-add), 

> Adds a component, in a specific configuration, to the current circuit template.
> 
> Format: `add CONFIG COMPONENT VALUE`
> 
> * The `CONFIG` can be chosen from the 2 - `series` and `parallel`.
> * The `COMPONENT` and `VALUE` are as explained above under [Set component value](https://github.com/AY2021S1-CS2113T-W13-3/tp/blob/master/docs/UserGuide.md#set).
> 
> Example of usage:
> 
> `add parallel c 500`

### LINKED ISSUE #6: Set Values of Components
Ability to set/change values of Resistors, Capacitors, Inductors so that calculations of new configurations can be done. Requires #3.

Based on the [User Guide section](https://github.com/AY2021S1-CS2113T-W13-3/tp/blob/master/docs/UserGuide.md#set-component-value-set-):

> Sets the value of the component. The component must be part of the current circuit template. Units correspond to the component involved - resistors in ohms, capacitors in microfarads, inductors in microhenries.
> 
> Format: `set COMPONENT VALUE`
> 
> * The `COMPONENT` can be chosen from the 3 - `r`, `c`, `l`
>     * `r` represents a resistor
>     * `c` represents a capacitor
>     * `l` represents an inductor
> * The `VALUE` can be an integer of a float, of which the units depends on the component set.
> 
> Example of usage:
> 
> `set r 500`

### LINKED ISSUE #22: Add execution to Command class, parsing to Parser class
For each command listed in the [User Guide](https://ay2021s1-cs2113t-w13-3.github.io/tp/UserGuide.html), there needs to be:

* parsing done in the `Parser` class to prepare the arguments
* a corresponding `Command` child class that can execute the command with its parameters.

Commands given:

* `help`
* `template`
* `set`
* `add`
* `calc`

Parsing was already done in #9.

### LINKED ISSUE #29: Add JUnit Testing
https://nus-cs2113-ay2021s1.github.io/website/schedule/week8/project.html#4-add-some-junit-tests

### PR DESCRIPTION
**Pull request by hughjazzman**
Closes #4, closes #6. Part of #22, #29.

* `template`, `add` and `set` commands works with ideal input.
* Refactor `Command` classes dealing with `Template`.
* Add `Template.setComponent()`.
* Add print statements in `Ui`.
* Complete `SetCommand` class.
* Use instance of `Parser.parse()`.
* Add JUnit tests.
* Update `DukeException` for invalid `Component` values.

### CODE REVIEW

File: src/main/java/seedu/duke/parser/Parser.java [hunk 1]
```diff
@@ -77,14 +85,20 @@ private static Command prepareTemplate(String[] args) throws DukeException {
...
     private static Template getTemplate(String arg) throws DukeException {
         switch (arg) {
         case "rl":
             return new LrTemplate();
+        case "rc":
+            return new RcTemplate();
+        case "r":
+            return new RTemplate();
+        case "lc":
+            return new LcTemplate();
```
Comment (sevenseasofbri) [inline src/main/java/seedu/duke/parser/Parser.java:101]: While I know having break statements out here is kind of useless but does this violate coding standard? https://se-education.org/guides/conventions/java/basic.html#layout

File: src/main/java/seedu/duke/parser/Parser.java [hunk 2]
```diff
@@ -94,17 +108,24 @@ private static Command prepareSet(String[] args) throws DukeException {
...
+        return new SetCommand(template, args[1], value);
     }
 
     private static Command prepareAdd(String[] args) throws DukeException {
         if (hasMinArguments(args, 4)) {
             throw new DukeException("Not enough arguments!");
         }
+        if (hasNoTemplate()) {
+            throw new DukeException("No template set yet!");
+        }
```
Comment (sevenseasofbri) [inline src/main/java/seedu/duke/parser/Parser.java:128]: Was going to ask about if the current template wasn't set, but saw this. This makes sense!

**Review by sevenseasofbri — APPROVED** [review]
LGTM!

File: src/main/java/seedu/duke/parser/Parser.java (hunk 1 above)
Comment (hughjazzman) [reply src/main/java/seedu/duke/parser/Parser.java:101]: Nope! In fact, it cannot compile if you add the `break` statements.

Comment (sevenseasofbri) [reply src/main/java/seedu/duke/parser/Parser.java:101]: Oh wow that's...interesting

File: src/main/java/seedu/duke/parser/Parser.java (hunk 2 above)
Comment (hughjazzman) [reply src/main/java/seedu/duke/parser/Parser.java:128]: We can consider adding more `Exception` classes and functionality later on.

**Pull request merged**